packages/core/src/hooks/auth/usePermissions/index.spec.ts

```diff
@@ -147,6 +180,31 @@ describe("v3LegacyAuthProviderCompatible usePermissions Hook", () => {
         result.current.refetch();
         expect(result.current.data).toBeUndefined();
     });
+
+    it("should accept params with v3LegacyAuthProviderCompatible", async () => {
+        const legacyGetPermissionMock = jest.fn(() => Promise.resolve(["admin"]));
+        const { result } = renderHook(
+            (props: PermissionsProps) => usePermissions({ ...props }),
+            {
+                initialProps: { params: 'admin', v3LegacyAuthProviderCompatible: true },
```

Comment:
```suggestion
                initialProps: { params: { currentRole: 'admin' }, v3LegacyAuthProviderCompatible: true },
```

Let's make this one object as mentioned before to cover more use-cases with minimal changes.